source/common/upstream/eds_subscription_factory.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "envoy/api/v2/core/base.pb.h"
+#include "envoy/config/subscription.h"
+#include "envoy/stats/scope.h"
+
+#include "common/config/eds_grpc_subscription_impl.h"
+#include "common/config/utility.h"
+#include "common/protobuf/protobuf.h"
+#include "common/config/grpc_mux_impl.h"
+#include "common/config/subscription_factory.h"
+
+namespace Envoy {
+namespace Upstream {
+  class EdsSubscriptionFactory {
+  public:
+    EdsSubscriptionFactory() : muxes_() {}
+
+    Config::GrpcMux& getOrCreateMux(const LocalInfo::LocalInfo& local_info, Grpc::AsyncClientPtr async_client,
```

Comment:
Should this be private? I don't think this is part of the interface.

Reply:
It's not part of the interface, but I have tests for it. The only thing I can think of in order to keep tests (which I would like to do) is to subclass ```EdsSubscriptionFactory``` in tests and add a proxy method for ```getOrCreateMux```. Would that work?

Reply:
Yeah, you can either use a testing peer pattern or provide a public `getOrCreateMuxForTests` that makes clear this is only to be used in tests (and make the existing implementation private and the target).